packages/components-react/projects/nextjs/tests/e2e/specs/modal-route-change.e2e.ts

```diff
@@ -0,0 +1,49 @@
+import type { Page } from 'puppeteer';
+import { goto, getAttribute, selectNode } from '../helpers';
+
+let page: Page;
+
+beforeEach(async () => (page = await browser.newPage()));
+afterEach(async () => await page.close());
+
+const pageUrl = '/modal-route-change';
+const getOpenModalBtn = () => selectNode(page, 'p-button >>> button');
+const getLinkToModal = () => selectNode(page, 'p-link a[href="/modal-route-change/open"]');
+const getDismissButton = () => selectNode(page, 'p-modal >>> p-button-pure >>> button');
+const getBodyStyle = async () => getAttribute(await selectNode(page, 'body'), 'style');
+
+it('should keep same scroll position when modal is opened and closed with route change', async () => {
```

Comment:
Test description tells a different story than actual implementation. It would be safer and clearer to actually check `scrollTop` values instead of relying on `.isIntersectingViewport()` especially since hit is what the title suggests? An element that is 1000px high only needs to be 1px in the viewport and this is true. If is entirely in the viewport it is true as well but scroll difference is at least 999px and probably more based on viewport height.